test-harness/tests/open/test_master.py

```diff
@@ -13,48 +13,60 @@
 
 log = logging.getLogger(__name__)
 
-
-class TestExhibitorEndpointOpen():
+authed_endpoints = ['/exhibitor',
+                    '/system/v1/agent/de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S1/logs/v1'
+                    '/system/v1/agent/de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S1/metrics/v0'
+                    '/system/v1/leader/marathon/foo/bar',
+                    '/system/v1/leader/mesos/foo/bar',
+                    '/system/v1/logs/v1',
+                    '/system/health/v1',
+                    '/system/v1/metrics',
+                    ]
+
+
+class TestAuthEnforcement():
+    @pytest.mark.parametrize("path", authed_endpoints)
     def test_if_unknown_user_is_forbidden_access(self,
                                                  master_ar_process,
-                                                 invalid_user_header):
+                                                 invalid_user_header,
+                                                 path):
         generic_unauthed_user_is_forbidden_test(master_ar_process,
                                                 invalid_user_header,
-                                                '/exhibitor/some/path')
+                                                path + "/foo/bar")
 
+    @pytest.mark.parametrize("path", authed_endpoints)
     def test_if_valid_user_is_permitted_access(self,
                                                master_ar_process,
-                                               valid_user_header):
+                                               valid_user_header,
+                                               path):
         generic_valid_user_is_permitted_test(master_ar_process,
                                              valid_user_header,
-                                             '/exhibitor/some/path')
+                                             path + "/foo/bar")
 
 
-class TestOpenSystemLoggingAgentEndpoint():
+class TestSystemAPILeaderProxingAuth():
     @pytest.mark.parametrize("path", [
-        (""),
-        ("/logs/v1"),
-        ("/metrics/v0"),
+        ("marathon"),
+        ("mesos"),
     ])
     def test_if_unknown_user_is_forbidden_access(self,
                                                  master_ar_process,
                                                  invalid_user_header,
                                                  path):
-        path_fmt = '/system/v1/agent/de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S1{}/foo/bar'
+        path_fmt = '/system/v1/leader/{}/foo/bar'
```

Comment:
Mhm, this diff seems to suggest that we actually change a test behavior, but that's just a bad luck, right?